modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]DefineAPISchemasWithinAnAPIGroup/CreateAPIApplicationSchema.testcase

```diff
@@ -0,0 +1,207 @@
+				title = "My-app");
+
+			ApplicationAPI.setUpGlobalAPIApplicationId();
+		}
+
+		task ("And Given I go to the Schemas tab of API Application edition") {
+			APIBuilderUI.switchToRelatedEntryInEditApplication(
+				entity = "Schemas",
+				title = "My-app");
+		}
+	}
+
+	tearDown {
+		ApplicationAPI.deleteAllAPIApplication();
+
+		JSONObject.deleteAllCustomObjects(pageSize = 22);
+
+		var testLiferayVirtualInstance = PropsUtil.get("test.liferay.virtual.instance");
+
+		if (${testLiferayVirtualInstance} == "true") {
+			PortalInstances.tearDownCP();
+		}
+	}
+
+	@priority = 5
+	test AllObjectDefinitionsAreAvailableInTheObjectMenu {
+		property portal.acceptance = "true";
+		property test.liferay.virtual.instance = "false";
+		property test.run.type = "single";
+
+		task ("Given multiple custom object definitions created without publishing") {
+			for (var index : list "1,2,3,4,5,6,7,8") {
+				ObjectDefinitionAPI.createObjectDefinition(
+					en_US_label = "Student${index}",
+					en_US_plural_label = "student${index}s",
+					name = "Student${index}",
+					requiredStringFieldName = "name");
+			}
+		}
+
+		task ("And Given all 20+ object deinifition entities available in Objects") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Object",
+				panel = "Control Panel",
+				portlet = "Objects");
+
+			AssertTextEquals(
+				locator1 = "Pagination#RESULTS",
+				value1 = "Showing 1 to 20 of 22 entries.");
+		}
+
+		task ("And Given one of object definitions on 2nd page") {
+			Click(locator1 = "DigitalSignatureListView#DIGITAL_SIGNATURE_NAVIGATE_NEXT");
+
+			AssertElementPresent(
+				key_name = "Student8",
+				locator1 = "ObjectAdmin#TABLE_LIST_TITLE");
+		}
+
+		task ("And Given I go to the Schemas tab of API Application edition") {
+			APIBuilderUI.switchToRelatedEntryInEditApplication(
+				entity = "Schemas",
+				title = "My-app");
+		}
+
+		task ("And Given I click Add New Schema") {
+			Click(locator1 = "APIBuilder#ADD_NEW_SCHEMA");
+		}
+
+		task ("When I click Object field") {
+			Click(locator1 = "APIBuilder#SELECT_AN_OBJECT_DEFINITION");
+		}
+
+		task ("Then all object deinifition entities from Objects are listed inluding the one placed on 2nd page of Objects") {
+			for (var objectDefinitionName : list "Address,AccountEntry,Organization,User,CommercePricingClass,CPDefinition,CommerceOrder,APISort,Bookmark,APIFilter,APIEndpoint,APIProperty,APISchema,APIApplication,Student1,Student2,Student3,Student4,Student5,Student6,Student7,Student8") {
+				AssertElementPresent(
+					key_kebabOption = ${objectDefinitionName},
+					locator1 = "ObjectAdmin#KEBAB_MENU_OPTION");
+			}
+		}
+	}
+
+	@priority = 5
+	test CanCreateMultipleSchemasInApplication {
+		property portal.acceptance = "true";
+		property test.liferay.virtual.instance = "false";
+		property test.run.type = "single";
+
+		task ("And Given a Schema created in the API Application") {
+			SchemaAPI.createAPISchema(
+				apiApplicationId = ${staticApplicationId},
```

Comment:
I wonder if instead of creating staticId we could create an Application with a specific ERC and use it to create Schema 🤔

Reply:
I think we can, and when I checked your PR, you have modified the macro in ApplciationAPI.macro, so I will move it to my PR to avoid conflicts

Reply:
Although erc is used at the .testcase level, erc is still used at the macro level to obtain the id and use the id to create the schema.

Reply:
maybe you can check my new changes, I used the application's ERC to create the schema and endpoint

Reply:
I just tested we can create a schema with r_apiApplicationToAPISchemas_c_apiApplicationERC, I will do more changes